server/src/test/java/org/opensearch/index/remote/RemoteStorePressureServiceTests.java

```diff
@@ -71,37 +68,14 @@ public void testIsSegmentsUploadBackpressureEnabled() {
         assertTrue(pressureService.isSegmentsUploadBackpressureEnabled());
     }
 
-    public void testAfterIndexShardCreatedForRemoteBackedIndex() {
-        IndexShard indexShard = createIndexShard(shardId, true);
-        pressureService = new RemoteStorePressureService(clusterService, Settings.EMPTY);
-        pressureService.afterIndexShardCreated(indexShard);
-        assertNotNull(pressureService.getRemoteRefreshSegmentTracker(indexShard.shardId()));
-    }
-
-    public void testAfterIndexShardCreatedForNonRemoteBackedIndex() {
-        IndexShard indexShard = createIndexShard(shardId, false);
-        pressureService = new RemoteStorePressureService(clusterService, Settings.EMPTY);
-        pressureService.afterIndexShardCreated(indexShard);
-        assertNull(pressureService.getRemoteRefreshSegmentTracker(indexShard.shardId()));
-    }
-
-    public void testAfterIndexShardClosed() {
-        IndexShard indexShard = createIndexShard(shardId, true);
-        pressureService = new RemoteStorePressureService(clusterService, Settings.EMPTY);
-        pressureService.afterIndexShardCreated(indexShard);
-        assertNotNull(pressureService.getRemoteRefreshSegmentTracker(shardId));
-
-        pressureService.afterIndexShardClosed(shardId, indexShard, indexShard.indexSettings().getSettings());
-        assertNull(pressureService.getRemoteRefreshSegmentTracker(shardId));
-    }
-
     public void testValidateSegmentUploadLag() {
         // Create the pressure tracker
         IndexShard indexShard = createIndexShard(shardId, true);
-        pressureService = new RemoteStorePressureService(clusterService, Settings.EMPTY);
-        pressureService.afterIndexShardCreated(indexShard);
+        remoteStoreStatsTrackerFactory = new RemoteStoreStatsTrackerFactory(Settings.EMPTY);
+        pressureService = new RemoteStorePressureService(clusterService, Settings.EMPTY, remoteStoreStatsTrackerFactory);
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
 
-        RemoteSegmentTransferTracker pressureTracker = pressureService.getRemoteRefreshSegmentTracker(shardId);
+        RemoteSegmentTransferTracker pressureTracker = pressureService.getRemoteSegmentTransferTracker(shardId);
```

Comment:
lets remove the method `getRemoteSegmentTransferTracker` from the pressure service and move all usages to tracker.

Reply:
Updated. Resolving this thread. Please feel free to re-open if needed.